app/Controllers/Http/ProfilesController.ts

```diff
@@ -0,0 +1,61 @@
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import UserProfile from 'App/Models/UserProfile'
+import UpdateProfileValidator from 'App/validators/UpdateProfileValidator'
+import Utils from 'App/utils/utils'
+import { Gender } from 'Contracts/Enums'
+import User from 'App/Models/User'
+
+export default class ProfilesController {
+  public async getProfile({ auth, response }: HttpContextContract) {
+    try {
+      const user = await User.query().where('id', auth.user?.id as number).preload('profile').first();
```

Comment:
Don't we get auth.user.id always if we have middleware, If we get then changes `auth.user?.id` => `auth.user.id` and also no need type cast